src/Handler/TagHandler.php

```diff
@@ -88,10 +101,34 @@ static function ($tag) use ($repoPrefix) {
                 },
                 $tags
             );
+
             // Also add a un-prefixed `ez-all` in order to be able to purge all across repos
             $tags[] = 'ez-all';
 
-            $response->headers->set($this->tagsHeader, implode(' ', array_unique($tags)));
+            $tagsString = implode(' ', array_unique($tags));
+            $tagsLength = strlen($tagsString);
+            if ($this->tagsHeaderMaxLength && $tagsLength > $this->tagsHeaderMaxLength) {
+                $tagsString = substr(
+                    $tagsString,
+                    0,
+                    // Seek backwards from point of max length using negative offset
+                    strrpos($tagsString, ' ', $this->tagsHeaderMaxLength - $tagsLength)
+                );
+
+                if (
+                    $this->tagsHeaderReducedTTl &&
+                    ($responseSharedMaxAge = $response->headers->getCacheControlDirective('s-maxage')) &&
+                    $this->tagsHeaderReducedTTl < $responseSharedMaxAge
+                ) {
+                    $response->setSharedMaxAge($this->tagsHeaderReducedTTl);
+                }
+
+                $this->logger->warning(
+                    'HTTP Cache tags header max length reached and truncated to ' . $this->tagsHeaderMaxLength
```

Comment:
Is there any way, which wouldn't overcomplicate implementation, to enrich log message with a list of removed tags so the user can potentially manually clear these tags by himself? I'm kinda aware that it's very unlikely, but if it doesn't hurt too much it would be great to have such information in the log.

Reply:
This is response and not purge*, so purging those tags at this point would be pointless (they are not anymore on this response so it won'1t be cleared)

_* purge is fine, we do chunking there and several request to backend, nothing lost there_

Reply:
Friday, 5 PM, please ignore 😄